include/dba.php

```diff
@@ -1299,37 +1299,29 @@ public static function buildCondition(array &$condition = [])
 						$condition_string .= " AND ";
 					}
 					if (is_array($value)) {
-						// Check if there are integer values in the parameters
+						/**
+						 * Workaround for MySQL Bug #64791.
+						 * Never mix data types inside any IN() condition.
+						 * In case of mixed types, cast all as string.
+						 * Logic needs to be consistent with dba::p() data types.
+						 */
 						$is_int = false;
-						$is_float = false;
 						$is_alpha = false;
 						foreach ($value as $single_value) {
 							if (is_int($single_value)) {
```

Comment:
This isn't just about integer, float value are concerned as well. If you do rewrite what @annando did earlier, please make it type agnostic. My first hunch would be to take the type of the first element of the array and check the type of the subsequent elements against it. At the first difference, the loop quits and casts everything as a string.